src/components/Tag/Tag.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { MdClose } from 'react-icons/md';
+import { bem } from '../../utils';
+import { Text } from '../Text';
+
+import styles from './Tag.scss';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+    /** Value that helps us to understand if the component is currently selected or not. */
+    isSelected?: boolean;
+    /** Color is assigned to a background color of a wrapper */
+    bgColor?: string;
+    /** Max-width of a component */
+    maxWidth?: string;
+    /** Size of the text */
+    size?: 'small' | 'normal' | 'large';
+    /** Callback, that is fired when a user clicks on a delete icon */
+    onDelete?: () => void;
+    /** Callback, that is fired when a user clicks on an element */
+    onClick?: () => void;
+}
+
+const { block, elem } = bem('Tag', styles);
+
+export const Tag: React.FC<Props> = (props) => {
+    const { children, bgColor, maxWidth, size, onDelete, onClick, isSelected, ...rest } = props;
+
+    // Generate title for children that are plain text (without tags)
+    // If there is something different from the string (JSX) - children will be of object type.
+    const areChildrenString = typeof children === 'string';
+
+    const handleDeleteClick = (e: React.MouseEvent<HTMLElement>) => {
+        e.stopPropagation();
+
+        onDelete?.();
+    };
+
+    return (
+        <div
+            {...block({ selected: isSelected, clickable: !!onClick, ...rest })}
```

Comment:
Nice job! This works great. Maybe we can simplify a bit and most importantly we need to apply `rest` to the `div`. This way we make sure additional attribute, like style for instance are correctly passed. So it would be:
`<div {...rest} {...bem({...props, clickable: !!onClick})}    ...  />`

Note that this will mean that the class you now call `selected` will need to be renamed in the scss file to `&--isSelected`.

Reply:
Implemented.

Reply:
Implemented.